tests/chainerx_tests/unit_tests/routines_tests/test_math.py

```diff
@@ -1104,6 +1104,42 @@ def test_log_softmax_invalid(device, a_shape, axis, dtype):
         return chainerx.log_softmax(a, axis=axis)
 
 
+@op_utils.op_test(['native:0', 'cuda:0'])
+@pytest.mark.parametrize('input', [
+    numpy.asarray(0), numpy.asarray(-1), numpy.asarray(1), numpy.asarray(
+        10), numpy.asarray(float('inf')), numpy.asarray(float('nan')),
+    numpy.full((), 2), numpy.full((0,), 2), numpy.full((2, 3), 2)
+])
+@pytest.mark.parametrize('contiguous', [None, 'C'])
+@pytest.mark.parametrize('in_dtype,out_dtype', _expected_dtypes_math_functions)
```

Comment:
Hm, I think your implementation aligning to NumPy resulting dtypes is fine which means `_expected_dtypes_math_functions` shouldn't be used here. It assumes certain types are promoted. Can you simply test all dtypes available in ChainerX and that the output has the same dtype?

Reply:
Sure! I'll do a test.